README.md

```diff
@@ -173,6 +173,38 @@ The Geolite2-City database can be downloaded from MaxMind's site at [https://dev
 Download the GeoIP2 Binary and save it at `geo_data/GeoLite2-City.mmdb`.
 The app will start using that Geolite2 file for geolocation after restart.
 
+#### Testing on a mobile device or in a virtual machine
+
+By default, the application binds to `localhost`. To test on a network device or within a virtual machine, you can bind to `0.0.0.0` instead, using the following instructions:
+
+1. Determine your computer's network IP address. On macOS, you can find this in the "Network" system settings, shown under the "Status: Connected" label. This often takes the format of `192.168.1.x` or `10.0.0.x`.
+2. In `config/application.yml`, replace `localhost` in the `domain_name` setting with the IP address discovered in the previous step. Leave the trailing port `:3000` unchanged.
+3. Start the server using the command `SMTP_HOST=192.168.1.x rails s -b 0.0.0.0`, replacing `SMTP_HOST` value with the IP address from the first step.
+  - Note that because this uses the Rails built-in server and bypasses the Foreman process, ancillary services like Mailcatcher and Webpack development server will not be started.
+4. Assuming that your phone or virtual machine computer is connected on the same network, visit the application using the domain name configured in the second step (for example, `http://192.168.1.131:3000`).
+
+#### Testing the application over HTTPS
+
+To browse the application locally over HTTPS, you will need to create a self-signed certificate:
+
+```
+mkdir -p tmp && openssl req -x509 -sha256 -nodes -newkey rsa:2048 -days 365 -keyout tmp/localhost.key -out tmp/localhost.crt
```

Comment:
Maybe we can include a script to run to generate this more easily? Especially if we can help automate populating the prompts.

I just stumbled on this: https://stackoverflow.com/a/8091488

At the very least, it might be preferable to include such a flag in the example command documented here.

Reply:
This, in combination with the Procfile changes is giving me an idea:

What if we added `make https` or `make run-https` and then add Makefile directives for those certs?

To detect local IP address maybe we could use something these [commands from StackOverflow](https://unix.stackexchange.com/a/318334) and then add a print step, like:

```
detected current IP... 123.456.789.000"
```

Reply:
A little bit of prior art for generating certs via Makefiles:

https://github.com/18F/identity-pki/blob/master/config/local-certs/Makefile

I don't think we have to worry about the Keychain trust stuff here, but the gist of "if this file doesn't exist, uses these commands to make it" is a great fit for Makefiles

Reply:
Yeah, I like that idea. It also touches on another subtlety of raising this for discussion, which is: It'd be one thing if a developer had to follow these steps once in an initial setup. But to have to do this every time I test Internet Explorer or on a phone is quite tedious. I've made a few aliases in my shell to help combat this, but I think we can be doing better to avoid needing to do that. Something like a `make run-https` would be a good improvement.